File: app/src/main/java/com/zwstudio/logicpuzzlesandroid/puzzles/holidayisland/domain/HolidayIslandObject.java
package com.zwstudio.logicpuzzlesandroid.puzzles.holidayisland.domain;

public abstract class HolidayIslandObject {
    public abstract String objAsString();
    public static HolidayIslandObject objFromString(String str) {
        switch (str) {
            case "marker":
                return new HolidayIslandMarkerObject();
            case "tree":
                return new HolidayIslandTreeObject();
            default:
                return new HolidayIslandEmptyObject();
        }
    }
}

File: app/src/main/java/com/zwstudio/logicpuzzlesandroid/puzzles/holidayisland/domain/HolidayIslandObject.kt
package com.zwstudio.logicpuzzlesandroid.puzzles.holidayisland.domain

abstract class HolidayIslandObject {
    abstract fun objAsString(): String

    companion object {
        fun objFromString(str: String?): HolidayIslandObject {
            return when (str) {
                "marker" -> HolidayIslandMarkerObject()
                "tree" -> HolidayIslandTreeObject()
                else -> HolidayIslandEmptyObject()
            }
        }
    }
}